src/oic/oic/__init__.py

```diff
@@ -808,17 +808,19 @@ def user_info_request(self, method="GET", state="", scope="", **kwargs):
         if "behavior" in kwargs:
             _behav = kwargs["behavior"]
             _token = uir["access_token"]
+            _ttype = ''
             try:
                 _ttype = kwargs["token_type"]
             except KeyError:
-                try:
-                    _ttype = token.token_type
-                except AttributeError:
-                    raise MissingParameter("Unspecified token type")
+                if token:
+                    try:
+                        _ttype = token.token_type
+                    except AttributeError:
+                        raise MissingParameter("Unspecified token type")
 
             if 'as_query_parameter' == _behav:
                 method = 'GET'
-            else:
+            elif token:
```

Comment:
* Comparison to `None` should be explicit.
* In tests, this condition is always true, so a test with token = `None` should be added.

Reply:
The comparison is still not explicit. It believe that it should be `elif token is not None:` as token is previously assigned `None`.

Reply:
I don't think that is good enough. token='' should be handled the same way as token=None
hence the use of ´<elif token>´ which implicitly include <token is not ''>.
So I'd rather not use not None since then I would have to have do
<elif token is not None and token is not ''>

Reply:
And is there any case where token would be an empty string?
If yes, then OK, leave the code as is.